project_info/git_utils.py

```diff
@@ -0,0 +1,107 @@
+import json
+import os
+import git
+import re
+import sys
+
+from git import GitCommandError
+
+from click import BadParameter, UsageError
+
+from .types import ReleaseType
+
+
+DEPLOYMENT_COMMIT_PATTERN = r'^Deployment of "(?P<branch_name>.+)"$'
+
+
+def create_release_branch(version, release_type, remote_name=None, branch_name=None):
+    repo = git.Repo(os.getcwd())
+    g = repo.git
+
+    if branch_name:
+        g.checkout(branch_name)
+    if remote_name:
+        g.pull(remote_name, branch_name)
+
+    if release_type in {ReleaseType.minor, ReleaseType.major}:
+        release_branch_name = 'release-{}'.format(version)
+    elif release_type == ReleaseType.patch:
+        release_branch_name = 'patch-{}'.format(version)
+    else:
+        raise BadParameter('build is not allowed for release')
+    g.checkout(branch_name or 'HEAD', b=release_branch_name)
+    return release_branch_name
+
+
+def create_deployment_branch(environment, remote_name=None, branch_name=None):
+    repo = git.Repo(os.getcwd())
+    g = repo.git
+    source_branch_name = repo.head.reference
+    deployment_branch_name = 'deploy-{}'.format(environment)
+
+    g.checkout('HEAD', b=deployment_branch_name)
+    g.commit('--allow-empty', message='Deployment of "{}"'.format(source_branch_name))
```

Comment:
I add release branch name to the message. It is easier to get the branch name for the merge to the next/master